src/CowWrapper.sol

```diff
@@ -0,0 +1,207 @@
+        uint256 sellTokenIndex;
+        uint256 buyTokenIndex;
+        address receiver;
+        uint256 sellAmount;
+        uint256 buyAmount;
+        uint32 validTo;
+        bytes32 appData;
+        uint256 feeAmount;
+        uint256 flags;
+        uint256 executedAmount;
+        bytes signature;
+    }
+
+    /// @notice Interaction data structure for pre/intra/post-settlement hooks
+    struct Interaction {
+        address target;
+        uint256 value;
+        bytes callData;
+    }
+
+    /// @notice Returns the authentication contract used by the settlement contract.
+    function authenticator() external view returns (ICowAuthentication);
+
+    /// @notice Returns the address of the vaultRelayer, the target for approvals for funds entering the settlement contract.
+    function vaultRelayer() external view returns (address);
+
+    /// @notice Returns the domain separator for EIP-712 signing
+    function domainSeparator() external view returns (bytes32);
+
+    /// @notice Allows for approval of orders by submitting an authorized hash on-chain prior to order execution.
+    function setPreSignature(bytes calldata orderUid, bool signed) external;
+
+    /// @notice Settles a batch of trades atomically
+    /// @param tokens Array of token addresses involved in the settlement
+    /// @param clearingPrices Array of clearing prices for each token
+    /// @param trades Array of trades to execute
+    /// @param interactions Array of three interaction arrays (pre, intra, post-settlement)
+    function settle(
+        address[] calldata tokens,
+        uint256[] calldata clearingPrices,
+        Trade[] calldata trades,
+        Interaction[][3] calldata interactions
+    ) external;
+}
+
+/// @title CoW Protocol Wrapper Interface
+/// @notice Interface for wrapper contracts that add custom logic around CoW settlements
+/// @dev Wrappers can be chained together to compose multiple settlement operations
+interface ICowWrapper {
+    /// @notice A human readable label for this wrapper. Used for display in explorer/analysis UIs
+    function name() external view returns (string memory);
+
+    /// @notice The settlement contract used by this wrapper
+    /// @return The CowSettlement contract address
+    function SETTLEMENT() external view returns (ICowSettlement);
+
+    /// @notice Initiates a wrapped settlement call
+    /// @dev This is the entry point for wrapped settlements. The wrapper will execute custom logic
+    ///      before calling the next wrapper or settlement contract in the chain.
+    /// @param settleData ABI-encoded call to ICowSettlement.settle() containing trade data
+    /// @param wrapperData Encoded data for this wrapper and the chain of next wrappers/settlement.
+    ///                    Format: [2-byte len][wrapper-specific-data][next-address]([2-byte len][wrapper-specific-data][next-address]...)
+    function wrappedSettle(bytes calldata settleData, bytes calldata wrapperData) external;
```

Comment:
`wrapperData` is also used in the function `parseWrapperData` immediately below but it's a different thing compared to the other one. We should use distinct names to avoid confusion like [this](https://github.com/cowprotocol/euler-integration-contracts/pull/13#discussion_r2582403270). Maybe something like:

```suggestion
    function wrappedSettle(bytes calldata settleData, bytes calldata chainedWrapperData) external;
```